packages/web-app-files/src/helpers/resource/move.ts

```diff
@@ -0,0 +1,220 @@
+        : $gettextInterpolate(
+            $gettext('Do this for all %{count} conflicts'),
+            { count: conflictCount },
+            true
+          ),
+      onCheckboxValueChanged: (value) => {
+        doForAllConflicts = value
+      },
+      onCancel: () => {
+        hideModal()
+        resolve({ strategy: ResolveStrategy.SKIP, doForAllConflicts } as ResolveConflict)
+      },
+      onConfirmSecondary: () => {
+        hideModal()
+        const strategy = ResolveStrategy.REPLACE
+        resolve({ strategy, doForAllConflicts } as ResolveConflict)
+      },
+      onConfirm: () => {
+        hideModal()
+        resolve({ strategy: ResolveStrategy.KEEP_BOTH, doForAllConflicts } as ResolveConflict)
+      }
+    }
+    createModal(modal)
+  })
+}
+export const resolveAllConflicts = async (
+  resourcesToMove,
+  targetFolder,
+  client,
+  createModal,
+  hideModal,
+  $gettext,
+  $gettextInterpolate,
+  resolveFileExistsMethod
+) => {
+  // if we implement MERGE, we need to use 'infinity' instead of 1
+  const targetFolderItems = await client.files.list(targetFolder.webDavPath, 1)
+  const targetPath = targetFolder.path
+  const index = targetFolder.webDavPath.lastIndexOf(targetPath)
+  const webDavPrefix = targetFolder.webDavPath.substring(0, index)
+
+  // Collect all conflicting resources
+  const allConflicts = []
+  for (const resource of resourcesToMove) {
+    const potentialTargetWebDavPath = join(webDavPrefix, targetFolder.path, resource.path)
+    const exists = targetFolderItems.some((e) => e.name === potentialTargetWebDavPath)
+    if (exists) {
+      allConflicts.push({
+        resource,
+        strategy: null
+      } as FileConflict)
+    }
+  }
+  let count = 0
+  let doForAllConflicts = false
+  let doForAllConflictsStrategy = null
+  const resolvedConflicts = []
+  for (const conflict of allConflicts) {
+    // Resolve conflicts accordingly
+    if (doForAllConflicts) {
+      conflict.strategy = doForAllConflictsStrategy
+      resolvedConflicts.push(conflict)
+      continue
+    }
+
+    // Resolve next conflict
+    const conflictsLeft = allConflicts.length - count
+    const result: ResolveConflict = await resolveFileExistsMethod(
+      createModal,
+      hideModal,
+      conflict.resource,
+      conflictsLeft,
+      $gettext,
+      $gettextInterpolate,
+      allConflicts.length === 1
+    )
+    conflict.strategy = result.strategy
+    resolvedConflicts.push(conflict)
+    count += 1
+
+    // User checked 'do for all x conflicts'
+    if (!result.doForAllConflicts) continue
+    doForAllConflicts = true
+    doForAllConflictsStrategy = result.strategy
+  }
+  return resolvedConflicts
+}
+export const showResultMessage = async (
+  errors,
+  movedResources,
+  showMessage,
+  $gettext,
+  $gettextInterpolate
+) => {
+  if (errors.length === 0) {
+    const count = movedResources.length
+    const title = $gettextInterpolate(
+      $gettext('%{count} item was moved successfully'),
```

Comment:
I fear we need to use `$ngettext()` here
<img width="536" alt="Screenshot 2022-06-02" src="https://user-images.githubusercontent.com/16822008/171756882-0d4e0479-79f7-44ce-82c2-aa5ea24406a2.png">

Reply:
done 👍🏼